applications/OptimizationApplication/custom_utilities/container_variable_data_holder/container_variable_data_holder.h

```diff
@@ -0,0 +1,157 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Suneth Warnakulasuriya
+//
+
+#pragma once
+
+// System includes
+#include <string>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+
+// Application includes
+#include "container_variable_data_holder_base.h"
+
+namespace Kratos {
+
+///@name Kratos Classes
+///@{
+
+template <class TContainerType, class TContainerDataIO>
+class KRATOS_API(OPTIMIZATION_APPLICATION) ContainerVariableDataHolder : public ContainerVariableDataHolderBase<TContainerType> {
+public:
+    ///@name Type definitions
+    ///@{
+
+    using BaseType = ContainerVariableDataHolderBase<TContainerType>;
+
+    using IndexType = std::size_t;
+
+    /// Pointer definition of ContainerVariableDataHolder
+    KRATOS_CLASS_POINTER_DEFINITION(ContainerVariableDataHolder);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor
+    ContainerVariableDataHolder(ModelPart& rModelPart)
+        : BaseType(rModelPart)
+    {
+    }
+
+    /// Copy constructor
+    ContainerVariableDataHolder(const ContainerVariableDataHolder& rOther)
+        : BaseType(rOther)
+    {
+    }
```

Comment:
This is unnecessary. The other constructor taking an instance of the baseclass does the same thing but is more general (+ it's safe in this case because the derived class doesn't have any extra member variables).

Reply:
removed